README.md

```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
 
-You can also combine multiple locations and use them as filters to increase the accuracy of your tests. For this purpose we use the plus symbol +.
+`globalping ping --help`
 
-- `from comcast+california` will return a probe that is both hosted with Comcast and located in CA.
-- `from google+europe` will return a random probe hosted with Google Cloud in Europe, or if you want to be more specific but dont remember the cloud region name:
-- `from google+germany` will return a probe with Google Cloud Germany
+###### The limit flag
+The `--limit` flag is probably one of the most useful flags when starting with Globalping. It determines the number of tests to run, which is set to one (1) by default. 
 
-You can combine as many parameters as you want too
+For example:
+- `from north america` selects one probe in either USA or Canada.
+- `from north america --limit 2` selects one probe in USA **and** one in Canada.
 
-- `from cogent+europe+datacenter` will return a probe hosted with Cogent, located in Europe and tagged as a datacenter probe. Or you could use `eyeball` to do the opposite.
+## Best practices and tips
+Learn to use Globalping in the most optimal way!
 
-### Multiple locations
+### Test with "magic" 🧙
+All integrations connect to the same API and use our "magic" field as location input.
+This maintains a consistent and straightforward user experience, allowing you to reuse logic and parameters across all tools.
 
-To select multiple locations in a single request you can use a comma. And of course you can combine filters and multiple locations at the same time.
+So, for example, when you run a test from "Germany" using the CLI or Slack app, our API's magic parser processes the location.
 
-- `from amazon+france,ovh+france --limit 2` will return a probe hosted with AWS in France and a probe hosted with OVH in France
+> [!Note]
+> Developers who want stricter and more predictable control over the selected probes and user input can use the individual location parameters [when making an API call](https://www.jsdelivr.com/docs/api.globalping.io#post-/v1/measurements), which expect you to provide each city, country, and tag in a standardized way.
 
-You can use this functionality to also define your own "world".
+The magic field supports a wide range of parameters and location combinations, including countries, continents, cities, US states, regions (Western Europe), ASNs, ISP names, eyeball or data center tags, and cloud region names (us-east-2).
 
-- `from germany, greece, uk, usa, canada, japan, china, south africa --limit 20` will return probes from all of the above countries in roughly equal proportions to fit the limit of 20. 
+### Re-select probes ♻️
+You can also provide the "magic" field with a measurement ID to have the API select the same probes used in a previous measurement. 
```

Comment:
Lets add something like:

 "This is a best effort action and if some of the probes are no longer online they will be missing from the new results. 
Additionally since measurements expire, after different intervals depending user type, you should not hard-code measurement IDs, as new tests will break after a measurement expires.
Make you sure you consider these limitations in your logic"